### PR TITLE
KEYCLOAK-2257: Store user attribute when accepting terms and conditions

### PR DESCRIPTION
**Pull request by raehalme**
Required action `TermsAndConditions` now stores user attribute indicating acceptance of terms and conditions.

### CODE REVIEW

File: services/src/main/java/org/keycloak/authentication/requiredactions/TermsAndConditions.java [hunk 1]
```diff
@@ -46,18 +50,22 @@ public void evaluateTriggers(RequiredActionContext context) {
...
     public void processAction(RequiredActionContext context) {
         if (context.getHttpRequest().getDecodedFormParameters().containsKey("cancel")) {
+            context.getUser().removeAttribute(USER_ATTRIBUTE);
             context.failure();
             return;
         }
-        context.success();
 
+        SimpleDateFormat dateTimeFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ");
+        context.getUser().setAttribute(USER_ATTRIBUTE, Arrays.asList(dateTimeFormat.format(new Date())));
```
Comment (stianst) [inline services/src/main/java/org/keycloak/authentication/requiredactions/TermsAndConditions.java]: We store time as seconds from 1970 everywhere else, as well as UserModel.getCreatedTimestamp so it should be changed to that to make it consistent (use org.keycloak.common.util.Time.currentTime)

**Comment by raehalme** [comment]
I did some quick searches to try to find out how other `String` representations for timestamp were implemented, but didn't catch that, sorry.

The number of seconds since 1970 kind of sucks in the UI under user attributes (that's why I chose the standard representation). Do you have any plans to improve this?

**Comment by stianst** [comment]
I'd like to at some point introduce something better for user attributes. What I had in mind was the ability to define a user profile, including standard attributes. For defined attributes you would be able to set which are required, verification and also input/output types. Based on that the admin console could display attributes differently. It's not something we'll be adding soon though.

**Comment by raehalme** [comment]
Ok, sounds good! I'll do the change you asked asap.

**Pull request merged**